src/Dependency/DependencyMap.php

```diff
@@ -34,39 +39,62 @@ public function __construct (array $dependencyMap = [])
      */
     public function getImportsWithDependencies (array $imports) : array
     {
+        /** @var AssetDependency[] $toLoad */
         $toLoad = [];
 
         foreach ($imports as $import)
         {
             $dirname = \dirname($import);
             $basename = \basename($import);
 
+            $dirname = ("." === $dirname)
+                ? ""
+                : "{$dirname}/";
+
             if ("js" === \pathinfo($basename, \PATHINFO_EXTENSION))
             {
-                $legacy = "{$dirname}/_legacy.{$basename}";
+                $legacy = "{$dirname}_legacy.{$basename}";
+                $modern = "{$dirname}_modern.{$basename}";
 
+                // load legacy Kaba builds: `entry` + `_legacy.entry`
                 if (\array_key_exists($import, $this->map) && \array_key_exists($legacy, $this->map))
                 {
-                    $toLoad = $this->loadForSingleImport($toLoad, $import, ["type" => "module"]);
-                    $toLoad = $this->loadForSingleImport($toLoad, $legacy, ["nomodule" => true]);
+                    $toLoad = $this->loadForSingleImport($toLoad, $import, self::MODERN);
+                    $toLoad = $this->loadForSingleImport($toLoad, $legacy, self::LEGACY);
+                    continue;
+                }
+
+                // load modern Kaba builds: `entry` + `_modern.entry`
+                if (\array_key_exists($import, $this->map) && \array_key_exists($modern, $this->map))
+                {
+                    $toLoad = $this->loadForSingleImport($toLoad, $import, self::LEGACY);
+                    $toLoad = $this->loadForSingleImport($toLoad, $modern, self::MODERN);
                     continue;
                 }
             }
 
             // load normally
-            $toLoad = $this->loadForSingleImport($toLoad, $import);
+            $toLoad = $this->loadForSingleImport($toLoad, $import, self::NEUTRAL);
         }
 
-        return \array_values($toLoad);
+        return \array_map(
+            function (AssetDependency $dependency)
+            {
+                return $dependency->getAssetEmbed();
+            },
+            $toLoad
+        );
     }
 
 
     /**
-     * @param string $import
+     * @param AssetDependency[] $allImports
+     * @param string            $import
+     * @param bool|null         $modernOrLegacy
      *
-     * @return AssetEmbed[]
+     * @return AssetDependency[]
      */
-    private function loadForSingleImport (array $allImports, string $import, array $embedAttributes = []) : array
+    private function loadForSingleImport (array $allImports, string $import, ?bool $modernOrLegacy) : array
```

Comment:
Having a boolean flag with an `or` in its name feels wrong to me, even if it's nullable. Can we either rename it to `bool $useModern` or change it to an `int` flag?

Reply:
`useModern` seems wrong, as these are 3 possible values.

I changed it to an `int`.